src/compiler/checker.ts

```diff
@@ -8646,39 +8646,36 @@ namespace ts {
           */
         function getEffectiveDecoratorFirstArgumentType(node: Node): Type {
             // The first argument to a decorator is its `target`.
-            switch (node.kind) {
-                case SyntaxKind.ClassDeclaration:
-                case SyntaxKind.ClassExpression:

```

Comment:
what happens with ClassExpression in the `if` version? it looks like it'll hit the Debug.fail